playbooks/os_updates.yml

```diff
@@ -63,20 +63,46 @@
         state: latest
       when: ansible_os_family == "OpenBSD"
 
-    - name: os_updates | Upgrade all packages
+    - name: Ubuntu | Upgrade all packages
       ansible.builtin.apt:
         update_cache: true
         upgrade: dist
         force_apt_get: true
       when: ansible_os_family == "Debian"
 
-    - name: os_updates | Check if reboot required
+    - name: RedHat | Upgrade all packages
+      ansible.builtin.yum:
+        name: "*" 
+        state: latest
+        update_cache: true
+        update_only: true
+      register: yum_update_status
+      when: ansible_os_family == "RedHat"
+
+    - name: RedHat | remove unneeded packages
+      ansible.builtin.yum:
+        autoremove: true
+      when: ansible_os_family == "RedHat"
+
+    - name: Ubuntu | Check if reboot required
       ansible.builtin.stat:
         path: /var/run/reboot-required
       register: reboot_required_file
       when: ansible_os_family == "Debian"
 
-    - name: os_updates | Reboot if required
+    - name: RedHat | Check if reboot required
+      ansible.builtin.reboot:
+        msg: "Reboot initiated by Ansible because of yum updates."
+        connect_timeout: 5
+        reboot_timeout: 600
+        pre_reboot_delay: 0
+        post_reboot_delay: 15
+        test_command: whoami
+      when: 
+        - ansible_os_family == "RedHat"
+        - yum_update_status.changed
```

Comment:
We need to register something here and then add a task to reboot the CentOS/RH boxes when required.

Reply:
I thought reboot happens when this register status changes. https://github.com/pulibrary/princeton_ansible/blob/be341d4f54912eb8c6254bbeb34e1596cfaa15a7/playbooks/os_updates.yml#L79

Reply:
Oh! You're absolutely right. I was reading the `name:` of the task and not what the task actually does.